source/dub/dependency.d

```diff
@@ -331,6 +359,10 @@ struct Dependency {
 
 				dep = Dependency.any;
 				dep.path = NativePath(verspec["path"].get!string);
+			} else if (auto repository = "repository" in verspec) {
+				enforce("version" in verspec, "No version field specified!");
+				dep = Dependency(Repository(repository.get!string),
```

Comment:
this should probably error out if repository is empty (see below)

Reply:
Added `enforce`.